### ISSUE TITLE
Rule proposal:  The file name cannot be index

### ISSUE THREAD

**Issue by laterdayi** [sindresorhus/eslint-plugin-unicorn#2047]
### Description

The file name cannot be index

error index.ts index.js index.png  index.vue

### Fail

```js
var replace = 'me';
```

```js
function foo() {
  var replace = 'me';
  return replace;
}
```

### Pass

```js
const replace = 'me';
```

```js
function foo() {
  return 'me';
}
```
